src/Symfony/Component/ExpressionLanguage/Node/ConditionalNode.php

```diff
@@ -48,4 +48,9 @@ public function evaluate($functions, $values)
 
         return $this->nodes['expr3']->evaluate($functions, $values);
     }
+
+    public function dump()
+    {
+        return sprintf('((%s) ? (%s) : (%s))', $this->nodes['expr1']->dump(), $this->nodes['expr2']->dump(), $this->nodes['expr3']->dump());
```

Comment:
I don't think the inner parenthesis are needed here as they are already added on other nodes when needed.